### PR TITLE
fix: allow pnpm to be used with next.js 9.5+

### PR DESCRIPTION
**Pull request by bitfrost**
Fix for https://github.com/vercel/next.js/issues/16471

### CODE REVIEW

File: packages/next/lib/resolve-request.ts [hunk 1]
```diff
@@ -8,6 +8,17 @@ export function resolveRequest(req: string, issuer: string): string {
...
+    // fixes webpack builds when using pnpm by switching to the default node resolution algorithm
+    // see: preserveSymlinks
+    // from : https://www.npmjs.com/package/resolve
+    //  - Note: this property is currently true by default but it will be changed to false in the next major
+    //           version because Node's resolution algorithm does not preserve symlinks by default.
+    const basedir =
+      issuer.endsWith(path.posix.sep) || issuer.endsWith(path.win32.sep)
+        ? issuer
+        : path.dirname(issuer)
+    return resolve.sync(req, { basedir, preserveSymlinks: false })
```
Comment (bitfrost) [inline packages/next/lib/resolve-request.ts:21]: We can also consider moving  preserveSymlinks: false  as default for all package managers and not have this conditional just for pnpm

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| buildDuration | 12.1s | 12.6s | ⚠️ +474ms |
| nodeModulesSize | 57.5 MB | 57.5 MB | ⚠️ +3.23 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.178 | 2.341 | ⚠️ +0.16 |
| / avg req/sec | 1147.63 | 1068.07 | ⚠️ -79.56 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.355 | 1.371 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1845.62 | 1823.64 | ⚠️ -21.98 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..7f47.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-da593e2..abf1.js gzip | 7.32 kB | 7.32 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.15 kB | 6.15 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-34d17c2..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-28298..e0c9.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-000f151..65d4.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.69 kB | 7.69 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-65c8a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-4cfda7a..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.35 kB | 5.35 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| index.html gzip | 972 B | 972 B | ✓ |
| link.html gzip | 979 B | 979 B | ✓ |
| withRouter.html gzip | 964 B | 964 B | ✓ |
| Overall change | 2.92 kB | 2.92 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.4s | -370ms |
| nodeModulesSize | 57.5 MB | 57.5 MB | ⚠️ +3.23 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..7f47.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-da593e2..abf1.js gzip | 7.32 kB | 7.32 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.15 kB | 6.15 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-34d17c2..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-28298..e0c9.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-000f151..65d4.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.69 kB | 7.69 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-65c8a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-4cfda7a..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.35 kB | 5.35 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _error.js | 1.03 MB | 1.03 MB | ✓ |
| 404.html | 4.22 kB | 4.22 kB | ✓ |
| hooks.html | 3.86 kB | 3.86 kB | ✓ |
| index.js | 1.03 MB | 1.03 MB | ✓ |
| link.js | 1.07 MB | 1.07 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.28 MB | 5.28 MB | ✓ |

</details>

</details>
Commit: 87e0974628d93061cc8edc1a52dbf7ca9ad42844

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| buildDuration | 11.7s | 11.5s | -296ms |
| nodeModulesSize | 57.5 MB | 57.5 MB | ⚠️ +3.23 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.225 | 2.03 | -0.2 |
| / avg req/sec | 1123.4 | 1231.78 | +108.38 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.155 | 1.094 | -0.06 |
| /error-in-render avg req/sec | 2165.17 | 2285.68 | +120.51 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..7f47.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-da593e2..abf1.js gzip | 7.32 kB | 7.32 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.15 kB | 6.15 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-34d17c2..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-28298..e0c9.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-000f151..65d4.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.69 kB | 7.69 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-65c8a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-4cfda7a..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.35 kB | 5.35 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| index.html gzip | 972 B | 972 B | ✓ |
| link.html gzip | 979 B | 979 B | ✓ |
| withRouter.html gzip | 964 B | 964 B | ✓ |
| Overall change | 2.92 kB | 2.92 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| buildDuration | 13.6s | 13.7s | ⚠️ +160ms |
| nodeModulesSize | 57.5 MB | 57.5 MB | ⚠️ +3.23 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..7f47.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-da593e2..abf1.js gzip | 7.32 kB | 7.32 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.15 kB | 6.15 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-34d17c2..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-28298..e0c9.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-000f151..65d4.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.69 kB | 7.69 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-65c8a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-4cfda7a..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.35 kB | 5.35 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _error.js | 1.03 MB | 1.03 MB | ✓ |
| 404.html | 4.22 kB | 4.22 kB | ✓ |
| hooks.html | 3.86 kB | 3.86 kB | ✓ |
| index.js | 1.03 MB | 1.03 MB | ✓ |
| link.js | 1.07 MB | 1.07 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.28 MB | 5.28 MB | ✓ |

</details>

</details>
Commit: c5a67ab95bf68401b18f4b086ead0daa89d28534

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| buildDuration | 12.1s | 12.3s | ⚠️ +164ms |
| nodeModulesSize | 57.5 MB | 57.5 MB | ⚠️ +3.23 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.22 | 2.256 | ⚠️ +0.04 |
| / avg req/sec | 1126.01 | 1108.17 | ⚠️ -17.84 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.307 | 1.289 | -0.02 |
| /error-in-render avg req/sec | 1913.4 | 1939.05 | +25.65 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..f20b.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-ce79c9a..12c9.js gzip | 7.36 kB | 7.36 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.13 kB | 6.13 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-9973c26..dule.js gzip | 6.42 kB | 6.42 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-28298..e0c9.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-000f151..65d4.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.69 kB | 7.69 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-65c8a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-4cfda7a..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.35 kB | 5.35 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| index.html gzip | 972 B | 972 B | ✓ |
| link.html gzip | 978 B | 978 B | ✓ |
| withRouter.html gzip | 964 B | 964 B | ✓ |
| Overall change | 2.91 kB | 2.91 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13.7s | ⚠️ +312ms |
| nodeModulesSize | 57.5 MB | 57.5 MB | ⚠️ +3.23 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..f20b.js gzip | 10.3 kB | 10.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-ce79c9a..12c9.js gzip | 7.36 kB | 7.36 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 57.3 kB | 57.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.13 kB | 6.13 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-9973c26..dule.js gzip | 6.42 kB | 6.42 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 52.3 kB | 52.3 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-9a0b9e1..b37e.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-28298..e0c9.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-89731c..c609.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-000f151..65d4.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.69 kB | 7.69 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-65c8a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-4cfda7a..dule.js gzip | 1.26 kB | 1.26 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.35 kB | 5.35 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | cvent/next.js fix_pnpm | Change |
| - | - | - | - |
| _error.js | 1.03 MB | 1.03 MB | ✓ |
| 404.html | 4.22 kB | 4.22 kB | ✓ |
| hooks.html | 3.86 kB | 3.86 kB | ✓ |
| index.js | 1.03 MB | 1.03 MB | ✓ |
| link.js | 1.07 MB | 1.07 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.28 MB | 5.28 MB | ✓ |

</details>

</details>
Commit: 75a45f830ccf875344e869e020ecba160e5e0bf4

**Comment by desmap** [comment]
@timneutkens @Timer pls review,+merge pnpm is light-years beyond limbo-state yarn, npm or lerna

File: packages/next/lib/resolve-request.ts [hunk 2]
```diff
@@ -8,6 +8,17 @@ export function resolveRequest(req: string, issuer: string): string {
   if (process.versions.pnp) {
     const { resolveRequest: pnpResolveRequest } = require(`pnpapi`)
     return pnpResolveRequest(req, issuer, { considerBuiltins: false })
+  } else if (process?.env?.npm_config_user_agent?.startsWith('pnpm/')) {
```
Comment (elliottsj) [inline packages/next/lib/resolve-request.ts:11]: Is this condition true only when running Next.js via `pnpm run <script>`? My concern is that this leads to false negatives: `node_modules` could be installed via pnpm, but we could be running Next.js via some other task runner, or directly by running `node_modules/.bin/next`.

File: packages/next/lib/resolve-request.ts (hunk 1 above)
Comment (elliottsj) [reply packages/next/lib/resolve-request.ts:21]: FWIW, this is the approach I took in https://github.com/vercel/next.js/pull/16369

File: packages/next/lib/resolve-request.ts (hunk 2 above)
Comment (bitfrost) [reply packages/next/lib/resolve-request.ts:11]: Yes, that would be a problem, it only fixes when running scripts via pnpm.  I would prefer your PR #16369 and what I mentioned above.    No conditional logic at all for pnpm.

**Comment by bitfrost** [comment]
Maintainers, please consider  #16369 instead.   If you would prefer conditional pnpm logic let me know and I can re-open and try a cleaner approach than npm_config_user_agent testing.

**Pull request closed without merging**